layouts/layouts/src/main/java/com/microsoft/device/dualscreen/layouts/FoldableLayoutController.kt

```diff
@@ -0,0 +1,361 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+package com.microsoft.device.dualscreen.layouts
+
+import android.content.Context
+import android.graphics.Rect
+import android.graphics.drawable.ColorDrawable
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.ViewGroup.LayoutParams.MATCH_PARENT
+import android.view.ViewGroup.LayoutParams.WRAP_CONTENT
+import android.widget.FrameLayout
+import android.widget.LinearLayout
+import android.widget.LinearLayout.HORIZONTAL
+import android.widget.LinearLayout.VERTICAL
+import androidx.annotation.LayoutRes
+import androidx.core.content.ContextCompat
+import androidx.core.view.doOnNextLayout
+import androidx.core.view.updateLayoutParams
+import androidx.window.layout.FoldingFeature
+import com.microsoft.device.dualscreen.utils.wm.ScreenMode
+import com.microsoft.device.dualscreen.utils.wm.getScreenRectangles
+import com.microsoft.device.dualscreen.utils.wm.getWindowRect
+import com.microsoft.device.dualscreen.utils.wm.normalizeWindowRect
+
+/**
+ * Class responsible for the logic of displaying the layout containers depending on the screen state.
+ * The class automatically handles the resize of the layout containers if the device rotates and
+ * also the position of the hinge.
+ */
+internal class FoldableLayoutController constructor(
+    private val rootView: FoldableLayout,
+    private var layoutConfig: FoldableLayout.Config
+) {
+    private val firstContainer = FrameLayout(rootView.context).apply {
+        id = R.id.first_container_id
+        layoutParams = LinearLayout.LayoutParams(MATCH_PARENT, MATCH_PARENT)
+    }.also {
+        rootView.addView(it)
+    }
+    private val secondContainer = FrameLayout(rootView.context).apply {
+        id = R.id.second_container_id
+        layoutParams = LinearLayout.LayoutParams(WRAP_CONTENT, MATCH_PARENT)
+    }.also {
+        rootView.addView(it)
+    }
+
+    var foldingFeature: FoldingFeature? = null
+        set(value) {
+            field = value
+            addContent(value.screenMode)
+        }
+
+    /**
+     * Add content depending on the given [ScreenMode]
+     * @param screenMode The given [ScreenMode]
+     */
+    private fun addContent(screenMode: ScreenMode) {
+        when (screenMode) {
+            ScreenMode.SINGLE_SCREEN -> addSingleScreenContent()
+            ScreenMode.DUAL_SCREEN -> addDualScreenContent()
+        }
+    }
+
+    /**
+     * Called when a new configuration change was requested.
+     *
+     * The function will take the containers inside [FoldableLayout]
+     * and change the width and height of them according to new [SurfaceDuoLayout.Config]
```

Comment:
Is this a typo? (SurfaceDuoLayout.Config) I guess maybe Foldable config?

Reply:
Done